src/main/java/com/google/sps/servlets/NewRecipeServlet.java

```diff
@@ -27,9 +17,41 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // implement new recipe
+    String recipeName = request.getParameter("recipeName");
+    String tagsResponse = request.getParameter("tags");
+    String ingredientsResponse = request.getParameter("ingredients");
+    String stepsResponse = request.getParameter("steps");
+    String privacy = request.getParameter("privacy");
+    
+    // splits the String responses from the form by commas/newlines and makes the resulting arrays into Lists
+    List<String> tags = new ArrayList<String>(Arrays.asList(tagsResponse.split(",", 0)));
+    List<String> ingredients = new ArrayList<String>(Arrays.asList(ingredientsResponse.split("\n", 0)));
+    List<String> steps = new ArrayList<String>(Arrays.asList(stepsResponse.split("\n", 0)));
+
+    // removes any empty, null, or newline members of the Lists
+    tags.removeAll(Arrays.asList("", null));
+    ingredients.removeAll(Arrays.asList("", null, "\n", "\r\n", "\r"));
+    steps.removeAll(Arrays.asList("", null, "\n", "\r\n", "\r"));
+
+    for (String tag : tags) {
+      tag.trim();
+    }
+    for (String ingredient : ingredients) {
+      ingredient.trim();
+    }
+    for (String step : steps) {
+      step.trim();
+    }
```

Comment:
I think you could do this as part of the splitting step above, and avoid iterating over all the lists. Here's a stackoverflow: https://stackoverflow.com/questions/41953388/java-split-and-trim-in-one-shot. You could do something like:
`List<String> tags = new ArrayList<String>(Arrays.asList(tagsResponse.trim().split("\\s*,\\s*");`
where you trim the original string, and then include whitespace in the split by using regular expressions. (if you're not familiar with regular expressions, here's a resource: https://medium.com/factory-mind/regex-tutorial-a-simple-cheatsheet-by-examples-649dc1c3f285. They're super handy for string parsing)

Or if you don't want to do that, take a look at Java 8 streams (see the second answer in that stackoverflow). It'll still be iterating over the lists but with much more concise (and in my opinion, more readable) syntax.